hs628_tools/feature_missing.py

```diff
@@ -0,0 +1,188 @@
+
+    def low_importance(self, features, target, target_type="classification", importance_thresh=1 / features.shape[1]):
+        # type: (ndarray, ndarray, str, float) -> ndarray
+
+        """
+        Find columns with their normalized importance value below pre-defined threshold using RandomForestRegressor or
+        RandomForestClassifier based on the target_type.
+            -----------
+            Parameters:
+                features :  numpy ndarrays, , shape = [n_samples, n_features]
+                            observations and features
+
+                target:     numpy ndarrays, shape = (1, ) target can be binary if target_type is classification or
+                            continuous if target_type is regression
+
+                target_type : string defines the RandomForest method if target is continuous or binary. default value
+                            = "classification"
+
+                importance_thresh : float between 0 and 1
+                            Pre-defined threshold set by the user, default value = 1/number of the features
+            -----------
+            Returns:
+                A ranked (sorted) list containing indices of features  with importance value below importance_thresh
+
+        """
+        # Performing the train_test_split
+        X_train, X_test, y_train, y_test = train_test_split(features, target, test_size=0.25)
+        # type: (ndarray, ndarray, ndarray, ndarray)
+        # Selecting the model based on target_type
+        if target_type == "regression":
+            model = RandomForestRegressor()
+        elif target_type == "classification":
+            model = RandomForestClassifier()
+        else:
+            raise ValueError('Target type must be "regression" or "classification" ')
+        # Fitting the model
+        model.fit(X_train, y_train)
+        feature_importance = model.feature_importances_
+        # normalizing the feature importance values
+        feat_imp_norm = feature_importance / np.sum(feature_importance)
+        # Creating a template (copy) to sort low important features
+        feat_imp_temp = feat_imp_norm.copy()
+        low_imp = []
+
+        for i in range(feat_imp_temp.shape[0]):
+            # finding the index of the minimum value and save it to low_importance
+            low_imp += [feat_imp_norm.argmin()]
+            # set the minimum to a large number in feat_imp_norm to find the next minimum value
+            feat_imp_norm[feat_imp_norm.argmin()] = 1000
+
+        low_importance = []
+        for i in range(len(low_imp)):
+            if feature_importance[low_imp[i]] < importance_thresh:
+                print ('Column #%d is #%d in the list of low importance features with the importnace value of %s \n' % (
+                low_imp[i], i + 1, feature_importance[low_imp[i]]))
+                low_importance +=[low_imp[i]]
+
+        self.low_importance_col = np.array(low_importance)
+        return self.low_importance_col
```

Comment:
Hi Omid, I reviewed your module and thought it was very well documented because of your clear and concise docstrings, which made reviewing your code way more efficient. Overall I believe this will be a very useful module!

Reply:
Great work Omid, it really is self explanatory the way you added all the comments : function definitions which are usually missed and overlooked. I really like the code structure. But I had the same question in mind as Nikita pointed out as well, How could you ever determine for a dataset which is a good threshold(40%,50%?) for missing or even correlation threshold(0.9) there could be cases when you have correlation value as 0.8 which is still a high correlation value and can say the features are colinear. Also if your dataset is indeed small then finding columns with greater than 40% missing values to eliminate/ignore them might not make sense. How about considering the size of the dataset for figuring out a threshold value ?

Reply:
Thank you @NikitaThomas :1st_place_medal: for reviewing my code and for your kind comment. To be honest, I don't know how to come up with a reasonable default value! I'll do more research and update my code. meanwhile, if you have any suggestion please send them my way.

Reply:
Thanks @rohitchadaram :1st_place_medal:  for taking time and reviewing my code. As we discussed int the class it might need more research on finding a reasonable default value for a parameter. I really like your idea of considering the size of the dataset as a driving factor to find the threshold. I'll try to find a way to incorporate that in my code.